src/Sulu/Bundle/SecurityBundle/Resources/config/doctrine/Role.orm.xml

```diff
@@ -11,7 +11,11 @@
             </cascade>
         </one-to-many>
         <one-to-many field="userRoles" target-entity="Sulu\Bundle\SecurityBundle\Entity\UserRole" mapped-by="role"/>
-        <one-to-many field="settings" target-entity="Sulu\Bundle\SecurityBundle\Entity\RoleSetting" mapped-by="role" index-by="key"/>
+        <one-to-many field="settings" target-entity="Sulu\Bundle\SecurityBundle\Entity\RoleSetting" mapped-by="role" index-by="key">
+            <cascade>
+                <cascade-remove/>
+            </cascade>
```

Comment:
Setting `on-delete="CASCADE"` on the join column of the other side is more performant, because the database is handling the deletion then on its own.